opm/simulators/flow/FlowProblemComp.hpp

```diff
@@ -438,13 +449,43 @@ class FlowProblemComp : public FlowProblem<TypeTag>
                     dofFluidState.setPressure(phaseIdx, pressure);
             }
 
-            for (unsigned compIdx = 0; compIdx < numComponents; ++compIdx) {
-                const std::size_t data_idx = compIdx * numDof + dofIdx;
-                const Scalar xmf = xmfData[data_idx];
-                const Scalar ymf = ymfData[data_idx];
+            if (has_xmf && has_ymf) {
+                const auto& xmfData = fp.get_double("XMF");
+                const auto& ymfData = fp.get_double("YMF");
+                for (unsigned compIdx = 0; compIdx < numComponents; ++compIdx) {
+                    const std::size_t data_idx = compIdx * numDof + dofIdx;
+                    const Scalar xmf = xmfData[data_idx];
+                    const Scalar ymf = ymfData[data_idx];
+
+                    dofFluidState.setMoleFraction(FluidSystem::oilPhaseIdx, compIdx, xmf);
+                    dofFluidState.setMoleFraction(FluidSystem::gasPhaseIdx, compIdx, ymf);
+                }
+            }
 
-                dofFluidState.setMoleFraction(FluidSystem::oilPhaseIdx, compIdx, xmf);
-                dofFluidState.setMoleFraction(FluidSystem::gasPhaseIdx, compIdx, ymf);
+            if (has_zmf) {
```

Comment:
This block doesn't do anything unless at least one of `gas_active` or `oil_active` is `true`.  Could there ever be a case where both of those are `false` at the same time?

Reply:
Although I would think it will be very strange, but I do not know whether it is allowed at all. I would add more checks inside the if condition.